bin/start.js

```diff
@@ -43,50 +41,66 @@ if (semver.lt(process.versions.node, '4.0.0')) {
   process.exit(1)
 }
 
-var knownOpts = {
-  help: Boolean,
-  version: Boolean,
-  loglevel: [
-    'silly',
-    'verbose',
-    'info',
-    'http',
-    'warn',
-    'error',
-    'silent'
-  ],
-  port: Number,
-  'bind-address': String,
-  public: path,
-  'in-memory': Boolean,
-  data: path,
-  'db-url': String
-}
-
-var shortHands = {
-  h: '--help',
-  usage: '--help',
-  v: '--version',
-  m: '--in-memory',
-  s: ['--loglevel', 'silent'],
-  d: ['--loglevel', 'info'],
-  dd: ['--loglevel', 'verbose'],
-  ddd: ['--loglevel', 'silly'],
-  silent: ['--loglevel', 'silent'],
-  verbose: ['--loglevel', 'verbose'],
-  quiet: ['--loglevel', 'warn']
-}
-
-var argv = nopt(knownOpts, shortHands)
-
-if (argv.help) {
-  process.stdout.write(fs.readFileSync(path.join(__dirname, 'readme.txt'), 'utf8'))
-  process.exit(0)
-}
-
-if (argv.version) {
+var options = yargs
+.options({
+  loglevel: {
+    choices: [
+      'silly',
+      'verbose',
+      'info',
+      'http',
+      'warn',
+      'error',
+      'silent'
+    ],
+    default: function () {
+      if ('s' in yargs.argv || 'ddd' in yargs.argv) return 'silly'
+      if ('d' in yargs.argv) return 'info'
+      if ('dd' in yargs.argv || 'verbose' in yargs.argv) return 'verbose'
+      if ('silent' in yargs.argv) return 'silent'
+      return 'warn'
+    }
+  },
+  port: {
+    type: 'number',
+    default: 6004
+  },
+  'bind-address': {
+    type: 'string',
+    default: '0.0.0.0'
+  },
+  public: {
+    type: 'string',
+    default: path.join(process.cwd(), 'public'),
+    normalize: true
+  },
+  m: {
+    alias: 'in-memory',
+    type: 'boolean',
+    default: true
+  },
+  data: {
+    type: 'string',
+    default: path.join(process.cwd(), '.hoodie'),
+    normalize: true
+  },
+  'db-url': {
+    type: 'string',
+    default: null
```

Comment:
is default required? Or can we leave it out? I think if we leave out `defualt:` it gets no listed in `--help`, but we can do `default: undefined`